lib/pure/asyncfutures.nim

```diff
@@ -30,7 +30,7 @@ type
 when not defined(release):
   var currentID = 0
 
-proc callSoon*(cbproc: proc ()) {.gcsafe.}
+var callSoon* {.threadvar.}: proc (cbproc: proc ()) {.gcsafe.}
```

Comment:
+1 It's good idea to use `{.threadvar.}` as in future this will allow different event loop for each thread.